### PR TITLE
fix: add Typescript types

### LINKED ISSUE #120: Add typescript support
I know its not very requested but it would be awesome to see type checking on this awesome lib :)

### PR DESCRIPTION
**Pull request by jaulz**
It would be great if someone could cross check these types 😊 

Should fix: https://github.com/fram-x/FluidTransitions/issues/120

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { NavigationRouteConfigMap, StackNavigatorConfig } from 'react-navigation';
```
Comment (jarvisluong) [inline lib/index.d.ts]:
```suggestion
import { NavigationRouteConfigMap, StackNavigatorConfig, NavigationContainer  } from 'react-navigation';
```

File: lib/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export { RouteDirection, Metrics, TransitionSpecification } from './Types';
+
+export function createFluidNavigator(
+  routeConfigMap: NavigationRouteConfigMap,
+  stackConfig: StackNavigatorConfig = {}
+): any;
```
Comment (jarvisluong) [inline lib/index.d.ts]:
```suggestion
): NavigationContainer;
```
as similar as in https://github.com/react-navigation/react-navigation/blob/542122794ad36d9ae68f33a453e31563147280bb/typescript/react-navigation.d.ts#L938

**Review by jarvisluong — CHANGES_REQUESTED** [review]
Awesome! There are just a couple of small changes that I proposed!

**Comment by jarvisluong** [comment]
@chrfalch I think this ts def is good, when would it be merged? I also have some project using ts and ultilizing this library 😁

**Comment by chrfalch** [comment]
Thanks, @jaulz and @jarvisluong ! I'll try to get this merged and published in the next few days!

**Comment by altruisticsoftware** [comment]
YES YES YES

**Comment by jaulz** [comment]
@chrfalch thanks!

**Pull request merged**